lib/iris/common/resolve.py

```diff
@@ -57,12 +64,260 @@
+        cube2.transpose()
+        cube3, cube4 = cube1, cube2
+        data = np.zeros(cube1.shape)
+        data1 = data * 10
+        data2 = data * 20
+        data3 = data * 30
+
+    .. doctest::
+
+        >>> print(cube1)
+        air_temperature / (K)               (time: 240; latitude: 37; longitude: 49)
+             Dimension coordinates:
+                  time                           x              -              -
+                  latitude                       -              x              -
+                  longitude                      -              -              x
+             Auxiliary coordinates:
+                  forecast_period                x              -              -
+             Scalar coordinates:
+                  forecast_reference_time: 1859-09-01 06:00:00
+                  height: 1.5 m
+             Attributes:
+                  Conventions: CF-1.5
+                  Model scenario: A1B
+                  STASH: m01s03i236
+                  source: Data from Met Office Unified Model 6.05
+             Cell methods:
+                  mean: time (6 hour)
+        >>> print(cube2)
+        air_temperature / (K)               (longitude: 49; latitude: 37)
+             Dimension coordinates:
+                  longitude                           x             -
+                  latitude                            -             x
+             Scalar coordinates:
+                  forecast_period: 10794 hours
+                  forecast_reference_time: 1859-09-01 06:00:00
+                  height: 1.5 m
+                  time: 1860-06-01 00:00:00, bound=(1859-12-01 00:00:00, 1860-12-01 00:00:00)
+             Attributes:
+                  Conventions: CF-1.5
+                  Model scenario: E1
+                  STASH: m01s03i236
+                  source: Data from Met Office Unified Model 6.05
+             Cell methods:
+                  mean: time (6 hour)
+        >>> resolver = Resolve(cube1, cube2)
+        >>> print(data.shape)
+        (240, 37, 49)
+        >>> result = resolver.cube(data)
+        >>> print(result)
+        air_temperature / (K)               (time: 240; latitude: 37; longitude: 49)
+             Dimension coordinates:
+                  time                           x              -              -
+                  latitude                       -              x              -
+                  longitude                      -              -              x
+             Auxiliary coordinates:
+                  forecast_period                x              -              -
+             Scalar coordinates:
+                  forecast_reference_time: 1859-09-01 06:00:00
+                  height: 1.5 m
+             Attributes:
+                  Conventions: CF-1.5
+                  STASH: m01s03i236
+                  source: Data from Met Office Unified Model 6.05
+             Cell methods:
+                  mean: time (6 hour)
```

Comment:
The doc-test equivalent of a picture to 1000 words... there is no need to describe what's happening to a detailed level... just show it. Works for me as a philosophy :wink: